src/addons/op-monitoring/src/main/java/org/niis/xroad/proxy/core/opmonitoring/OpMonitoringDataProcessor.java

```diff
@@ -96,4 +100,27 @@ private static boolean isNonLoopback(NetworkInterface ni) {
             throw XrdRuntimeException.systemException(e);
         }
     }
+
+    private static boolean hasAnyUsableAddress(NetworkInterface ni) {
+        return ni.inetAddresses().anyMatch(OpMonitoringDataProcessor::isUsableAddress);
+    }
+
+    private static int compareNetworkInterfaces(NetworkInterface ni1, NetworkInterface ni2) {
+        return Integer.compare(ni1.getIndex(), ni2.getIndex());
+    }
+
+    private static boolean isUsableAddress(InetAddress addr) {
+        return !addr.isLoopbackAddress()
+                && !addr.isLinkLocalAddress()
+                && addr.getHostAddress() != null;
+    }
+
+    private static int compareInetAddresses(InetAddress a1, InetAddress a2) {
+        // prefer IPv4 addresses
+        return Boolean.compare(a1.getHostAddress().contains(":"), a2.getHostAddress().contains(":"));
```

Comment:
Maybe using instanceof check instead of string search would be more reliable?

```suggestion
        return Boolean.compare(a1 instanceof Inet6Address, a2 instanceof Inet6Address);
```